docs/plugins.rst

```diff
@@ -0,0 +1,126 @@
+downloaded or if the version matches one of the currently installed compiler versions. 
+The contracts are then grouped by compiler type and version and fed into the corresponding compiler to compile them. 
+These are then output as a JSON file to the ``.build`` directory. They can then be deployed on the chain from the console or a script.
+
+
+Source type
+===========
+Types from types module :doc:`source types <autoapi/ape/types/contract/index.html#ape.types.contract.Source>`
+
+Compiler manager
+================
+The compiler manager contains all the registered compilers. 
+Compiler plugins must subclass the `CompilerAPI <autoapi/ape/api/compiler/index#ape.api.compiler.CompilerAPI>`_ object and implement all ``abstractmethod``.
+Implement `get_versions <autoapi/ape/api/compiler/index#ape.api.compiler.CompilerAPI.get_versions>`_ in compile.
+``get_versions`` gets a set of all the files and tell it all the versions that are needed. 
+It needs to get that information to create the manifest so it can record the compilers which are required. 
+From the contract types you can then initialize or deploy a contract using the contract types.
+data structure source is how to get the source file from the manifest
+This method should always return the same value and doesn't cache.
+Compiler Manager uses the compiler api and has a list of all the compiler api subclasses. 
+The compiler manager has the set of all the registered compiler plugins. 
+Those compiler plugins subclass the compiler api, 
+and so that's how it can call out to the plugins in order to compile files 
+which are detected inside the contracts folder.
+
+CompilerAPI plugins
+
+ContractType type
+==================
+The compilation produces the __.build
+The manifest is a file that describes the package. 
+It describes everything that is within the package. 
+The package manifest contains links to source code on ipfs, 
+a heirarchy of which compilers and which versions of those compilers are used to compile files, 
+contract types which come from the compiled files that you might want to use in the package.
+`ContractType <autoapi/ape/types/contract/index.html#ape.types.contract.ContractType>`_
+
+
+Account registration process flow
+=================================
+accounts manager
+AccountAPI/AccountContainerAPI plugins
+Signing messages and transactions via AccountAPI
+
+
+Transactional process flow
+==========================
+networks manager
+EcosystemAPI plugins
+NetworkAPI plugins
+ProviderAPI plugins
+ExplorerAPI plugins
+ContractInstance type and encoding via EcosystemAPI
+
+
+Test process flow
+=================
+This is currently under development. We will have more documenting surrounding testing as it becomes completed.
+Writing fixtures
+TBD...
+
+
+Argument conversion process flow
+================================
+CLI arguments are decoded and passed in to the application with ``click``.
+
+Plugin Heirarchy
```

Comment:
Could re remove this Plugin hierarchy TBD section since it is already defined above?

Reply:
removed this erroneous section